var/spack/repos/builtin/packages/exciting/package.py

```diff
@@ -0,0 +1,121 @@
+
+    for __compiler in spack.compilers.supported_compilers():
+        if __compiler != 'intel':
+            conflicts('%{0}'.format(__compiler), when='^mkl',
+                      msg='MKL only works with the Intel compiler')
+
+    def edit(self, spec, prefix):
+        opts = {}
+        opts['BUILDSMP'] = 'true'
+        opts['F90_OPTS'] = '-cpp '
+        opts['F77_OPTS'] = '-cpp -O3 '
+        opts['CPP_ON_OPTS'] = '-cpp -DXS -DISO -DLIBXC'
+        opts['LIB_ARP'] = 'libarpack.a'
+        opts['F90'] = spack_fc
+        opts['F77'] = spack_f77
+        if '+omp' in spec:
+            opts['LDFLAGS'] = self.compiler.openmp_flag + ' -DUSEOMP'
+            opts['F90_OPTS'] += self.compiler.openmp_flag + ' -DUSEOMP'
+            opts['F77_OPTS'] += self.compiler.openmp_flag + ' -DUSEOMP'
+        if '%intel' in spec:
+            opts['F90_OPTS'] += ' -O3 -cpp -ip -unroll -scalar_rep '
+            opts['CPP_ON_OPTS'] += ' -DIFORT -DFFTW'
+        if '%gcc' in spec:
+            opts['F90_OPTS'] += '-O3 -march=native -ffree-line-length-0'
+
+        filter_file('FCFLAGS = @FCFLAGS@',
+                    ' '.join(['FCFLAGS = @FCFLAGS@', '-cpp',
+                              self.compiler.openmp_flag]),
+                    'src/libXC/src/Makefile.in')
+        if '+mkl' in spec:
+            if '%intel' in spec:
+                opts['LIB_LPK'] = '-mkl=parallel'
+            opts['INC_MKL'] = spec['mkl'].headers.include_flags
+            opts['LIB_MKL'] = spec['mkl'].libs.ld_flags
+        else:
+            opts['LIB_LPK'] = ' '.join([spec['lapack'].libs.ld_flags,
+                                        spec['blas'].libs.ld_flags,
+                                        self.compiler.openmp_flag])
+        if '+mpi' in spec:
+            opts['BUILDMPI'] = 'true'
+            opts['MPIF90'] = spec['mpi'].mpifc
+            opts['MPIF90_CPP_OPTS'] = self.compiler.openmp_flag
+            opts['MPIF90_CPP_OPTS'] += ' -DMPI -DMPIRHO -DMPISEC '
+            opts['MPIF90_OPTS'] = ' '.join(['$(F90_OPTS)', '$(CPP_ON_OPTS) '
+                                            '$(MPIF90_CPP_OPTS)'])
+            opts['MPIF90MT'] = '$(MPIF90)'
+        else:
+            opts['BUILDMPI'] = 'false'
+
+        if '+scalapack' in spec:
+            opts['LIB_SCLPK'] = spec['scalapack'].libs.ld_flags
+            opts['LIB_SCLPK'] += ' ' + self.compiler.openmp_flag
+            opts['CPP_SCLPK'] = ' -DSCAL '
+            opts['LIBS_MPI'] = '$(LIB_SCLPK)'
+            opts['MPIF90_CPP_OPTS'] += ' $(CPP_SCLPK) '
+
+        opts['USE_SYS_LAPACK'] = 'true'
+        opts['LIB_FFT'] = 'fftlib.a'
+        opts['LIB_BZINT'] = 'libbzint.a'
+        opts['LIBS'] = '$(LIB_ARP) $(LIB_LPK) $(LIB_FFT) $(LIB_BZINT)'
+        with open('build/make.inc', 'a') as inc:
+            for key in opts:
+                inc.write('{0} = {1}\n'.format(key, opts[key]))
+
+    def install(self, spec, prefix):
+        cd = which('cd')
+        make()
```

Comment:
You'll want to do this in the `build` method, not in `install`. The build method already runs `make` for you, so you can probably remove this.

Reply:
right!  fixed now.